apps/src/lib/levelbuilder/reference-guide-editor/ReferenceGuideEditAll.jsx

```diff
@@ -0,0 +1,222 @@
+
+  // To move guides, we will swap the positions of guides in the direction we want to move.
+  // The positions might sometimes have gaps (in the case of deletion or changing parents),
+  // but if we swap position values, they'll still sort correctly.
+  // Using this strategy, we don't need to update the positions for everything in the list.
+  // `direction` should be -1 or 1
+  const moveGuide = useCallback(
+    (
+      {key: targetGuideKey, parent_reference_guide_key: targetParentKey},
+      direction
+    ) => {
+      const guideSiblings = getGuideChildren(targetParentKey, referenceGuides);
+      const targetGuideIndex = guideSiblings.findIndex(
+        guide => guide.key === targetGuideKey
+      );
+      const swapGuideIndex = targetGuideIndex + direction;
+      if (swapGuideIndex < 0 || swapGuideIndex >= guideSiblings.length) {
+        // we don't need up update anything if it is at the top or bottom of the list
+        return;
+      }
+
+      // the target guide is the one we clicked the move button on,
+      // and the swap guide is the one in the direction we want to move
+      const targetGuide = guideSiblings[targetGuideIndex];
+      const swapGuide = guideSiblings[swapGuideIndex];
+
+      // update the local list
+      const swapPosition = swapGuide.position;
+      swapGuide.position = targetGuide.position;
+      targetGuide.position = swapPosition;
+      setReferenceGuides([...referenceGuides]);
+
+      // update the db (using the updated positions)
+      const targetUpdate = fetch(`${baseUrl}/${targetGuide.key}`, {
+        method: 'PUT',
+        headers: {
+          'content-type': 'application/json',
+          'X-CSRF-Token': $('meta[name="csrf-token"]').attr('content')
+        },
+        body: JSON.stringify({
+          position: targetGuide.position
+        })
+      });
+      const swapUpdate = fetch(`${baseUrl}/${swapGuide.key}`, {
+        method: 'PUT',
+        headers: {
+          'content-type': 'application/json',
+          'X-CSRF-Token': $('meta[name="csrf-token"]').attr('content')
+        },
+        body: JSON.stringify({
+          position: swapGuide.position
+        })
+      });
+      return Promise.all([targetUpdate, swapUpdate]);
+    },
+    [referenceGuides, baseUrl]
+  );
+
+  return (
+    <div>
+      <h1>Reference Guides</h1>
+      <div className="page-actions">
+        <TextLink
+          className="create-btn"
+          icon={<FontAwesome icon="plus" />}
+          iconBefore={true}
+          text="Create New Guide"
+        />
+      </div>
+
+      {showDeleteWarningDialog && (
+        <Dialog
+          body={
+            <>
+              <DialogTitle>{`Are you sure you want to permanently delete reference guide ${pendingDeleteKey}?
+          `}</DialogTitle>
+              <DialogBody>
+                <p>
+                  This will also delete any child reference guides. List of
```

Comment:
nit: Instead of "List of guides affected" maybe "The following child reference guides will be deleted:"